tests/benchmark/custom_memory_manager/include/custom_memory_manager.hpp

```diff
@@ -0,0 +1,117 @@
+
+    /// @brief Custom std::realloc wrapper which registers the re-allocation in a global BenchmarkMemoryManager object
+    /// @param[ptr] Pointer to the memory area to be reallocated
+    /// @param [new_size] New size of the array
+    /// @return Pointer to the beginning of newly allocated memory
+    void* CustomMemoryManager::Realloc(void* ptr, std::size_t new_size);
+
+    /// @brief Custom _aligned_malloc (WIN32) or std::aligned_alloc (LINUX) wrapper
+    ///        which registers the allocation in a global BenchmarkMemoryManager object
+    /// @param[size] Number of bytes of uninitialized storage to be allocated
+    /// @param[alignment] Specifies the alignment
+    /// @return The pointer to the beginning of newly allocated memory
+    void* CustomMemoryManager::AlignedAlloc(size_t size, size_t alignment);
+
+    /// @brief Gets the total number of allocations
+    /// @return Number of allocations
+    int64_t Allocations() const;
+
+    /// @brief Gets the total number of deallocations
+    /// @return Number of deallocations
+    int64_t Deallocations() const;
+
+    /// @brief Gets the total memory allocated in bytes between Start and Stop
+    /// @return Total memory allocated in bytes
+    int64_t TotalAllocatedBytes() const;
+
+    /// @brief Gets the peak memory use in bytes between Start and Stop
+    /// @return Peak memory use in bytes between Start and Stop
+    int64_t MaxBytesUsed() const;
+
+    /// @brief Gets the net changes in memory in bytes between Start and Stop
+    /// @return Net changes in memory in bytes between Start and Stop
+    int64_t NetHeapGrowth() const;
+
+    /// @brief Checks if there are leaks based on the net between the number of allocations
+    ///        and deallocations or the net heap growth
+    /// @return Boolean indicating whether a leak has been detected
+    bool HasLeaks() const;
+
+    /// @brief Resets the memory statistics
+    void ResetStatistics();
+
+    friend std::ostream& operator<<(std::ostream& ostream, CustomMemoryManager const& custom_memory_manager);
+
+    std::string Statistics(std::string const& caller = std::string()) const;
+
+private:
+    int64_t m_num_allocations = 0;             ///< The number of allocations made in total between Start and Stop
+    int64_t m_num_deallocations = 0;           ///< The number of deallocations made in total between Start and Stop (not written to ::benchmark::MemoryManager::Result)
+    int64_t m_total_allocated_bytes = 0;       ///< The total memory allocated in bytes between Start and Stop
+    int64_t m_max_bytes_used = TombstoneValue; ///< The peak memory use in bytes between Start and Stop
+    int64_t m_net_heap_growth = 0;             ///< The net changes in memory in bytes between Start and Stop
+
+    CustomMemoryManager() = default;
+    ~CustomMemoryManager() = default;
+    CustomMemoryManager(CustomMemoryManager const&) = delete;
+    CustomMemoryManager& operator=(CustomMemoryManager const&) = delete;
```

Comment:
I thought they're implicitly deleted here. I'll double check.

Reply:
It seems like what I thought is true. From the [move constructors doc](https://en.cppreference.com/w/cpp/language/move_constructor)
```
If no user-defined move constructors are provided for a class type (struct, class, or union), and all of the following is true:

1. there are no user-declared copy constructors;
2. there are no user-declared copy assignment operators;
3. there are no user-declared move assignment operators;
4. there is no user-declared destructor. 

then the compiler will declare a move constructor as a non-explicit inline public member of its class with the signature T::T(T&&). 
```
Items 1 and 2 are not met because the copy ctorand the copy assignment op are explicitly deleted. So deleting the move ctors is not really necessary.